config/crd/snapshot.storage.k8s.io_volumesnapshots.yaml

```diff
@@ -0,0 +1,141 @@
+                If both source and volumeSnapshotContentName are not specified, dynamic
+                snapshot creation will fail.
+              properties:
+                apiGroup:
+                  description: APIGroup is the group for the resource being referenced.
+                    If APIGroup is not specified, the specified Kind must be in the
+                    core API group. For any other third-party types, APIGroup is required.
+                  type: string
+                kind:
+                  description: Kind is the type of resource being referenced
+                  type: string
+                name:
+                  description: Name is the name of resource being referenced
+                  type: string
+              required:
+              - kind
+              - name
+              type: object
+            volumeSnapshotClassName:
+              description: 'volumeSnapshotClassName is the name of the VolumeSnapshotClass
+                requested by the VolumeSnapshot. If not specified, the default snapshot
+                class will be used if there exists one. If not specified, and there
+                is no default snapshot class, dynamic snapshot creation will fail.
+                More info: https://kubernetes.io/docs/concepts/storage/volume-snapshot-classes'
+              type: string
+            volumeSnapshotContentName:
+              description: volumeSnapshotContentName is the name of the VolumeSnapshotContent
+                object backing the VolumeSnapshot. To request a pre-existing VolumeSnapshotContent
+                object, this field MUST be specified during VolumeSnapshot object
+                creation. Otherwise, a snapshot will be dynamically created, and this
+                field will be populated afterwards.
+              type: string
+          type: object
+        status:
+          description: 'status represents the current information of a snapshot. NOTE:
+            status can be modified by sources other than system controllers, and must
+            not be depended upon for accuracy. Controllers should only be using information
+            from the VolumeSnapshotContent object after verifying that the binding
+            is accurate and complete.'
+          properties:
+            creationTime:
+              description: 'creationTime, if not nil, represents the timestamp when
+                the snapshot was successfully cut by the underlying storage system.
+                When dynamically creating a snapshot, this will be filled in upon
+                creation. For pre-existing snapshot, it will be filled in upon the
+                VolumeSnapshot object has been successfully bound to a VolumeSnapshotContent
+                object and the underlying storage system have the information available.
+                NOTE: Controllers MUST NOT rely on this field programmatically.'
+              format: date-time
```

Comment:
Should the format match the format in VolumeSnapshotContent?

Reply:
I am not sure whether these two fields (creationTime/restoreSize) should be the same in the VolumeSnapshotContent and in VolumeSnapshot.Status
the content object holds the value returned directly from CSI call, both are int64 type, where in snapshot.Status, it holds K8s types (datetime for creationTime, and Quantity for restoreSize)
As the spec, I would think it might make sense to keep whatever format returned directly? and for status, its easier for users to understand.